rllib/policy/sample_batch.py

```diff
@@ -458,7 +458,7 @@ def finish_slice():
         # For each unique env timestep.
         for _, group in itertools.groupby(steps, lambda x: x[:2]):
             # Accumulate into the current slice.
```

Comment:
This only has the effect of sorting by agent index as well, correct?

Reply:
That's correct. It is needed to sort the list because the tuple (episode_id, timestep, policy) is no longer unique and there is no order for dictionaries.